client.go

```diff
@@ -82,22 +83,34 @@ func NewDgraphClient(clients ...api.DgraphClient) *Dgraph {
 //		defer conn.Close()
 //		dgraphClient := dgo.NewDgraphClient(api.NewDgraphClient(conn))
 func DialCloud(endpoint, key string) (*grpc.ClientConn, error) {
-	u, err := url.Parse(endpoint)
-	if err != nil {
-		return nil, err
+	var grpcHost string
+	switch {
+	case strings.Contains(endpoint, ".grpc.") && strings.Contains(endpoint, ":"+cloudPort):
+		grpcHost = endpoint
+	case strings.Contains(endpoint, ".grpc.") && !strings.Contains(endpoint, ":"+cloudPort):
+		grpcHost = endpoint + ":" + cloudPort
+	default:
+		if !strings.HasPrefix(endpoint, "http") {
```

Comment:
Can we verify that it starts with http or https?

Reply:
https starts with http, no need